crates/ra_assists/src/handlers/number_representation.rs

```diff
@@ -0,0 +1,597 @@
+use ra_syntax::{ast, ast::LiteralKind, AstNode, SmolStr};
+use std::fmt;
+
+use crate::{Assist, AssistCtx, AssistId};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+enum NumberLiteralType {
+    /// A literal without prefix, '42'
+    Decimal,
+    /// Hexadecimal literal, '0x2A'
+    Hexadecimal,
+    /// Octal literal, '0o52'
+    Octal,
+    /// Binary literal, '0b00101010'
+    Binary,
+}
+
+#[derive(Clone, Debug)]
+struct NumberLiteral {
+    /// The type of literal (no prefix, hex, octal or binary)
+    number_type: NumberLiteralType,
+    /// The suffix as a string, for example 'i32'
+    suffix: Option<SmolStr>,
+    /// The prefix as string, for example '0x'
+    prefix: Option<SmolStr>,
+    /// Text of the literal
+    text: String,
+}
+
+impl fmt::Display for NumberLiteral {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Some(prefix) = &self.prefix {
+            f.write_str(prefix)?;
+        }
+
+        f.write_str(&self.text)?;
+
+        if let Some(suffix) = &self.suffix {
+            f.write_str(suffix)?;
+        }
+
+        Ok(())
+    }
+}
+
+fn identify_number_literal(literal: &ast::Literal) -> Option<NumberLiteral> {
+    match literal.kind() {
+        LiteralKind::IntNumber { suffix } => {
+            let token = literal.token();
+            let full_text = token.text().as_str();
+            let suffix_clone = suffix.clone();
+            let suffix_len = suffix.map(|s| s.len()).unwrap_or_default();
+            let non_suffix = &full_text[0..full_text.len() - suffix_len];
+            let maybe_prefix = if non_suffix.len() < 2 { None } else { Some(&non_suffix[0..2]) };
+            let (prefix, number_type) = match maybe_prefix {
+                Some("0x") => (maybe_prefix, NumberLiteralType::Hexadecimal),
+                Some("0b") => (maybe_prefix, NumberLiteralType::Binary),
+                Some("0o") => (maybe_prefix, NumberLiteralType::Octal),
+                _ => (None, NumberLiteralType::Decimal),
+            };
+            let prefix_len = prefix.map(|s| s.len()).unwrap_or_default();
+            let text = &non_suffix[prefix_len..];
+
+            let result = NumberLiteral {
+                number_type,
+                suffix: suffix_clone,
+                prefix: prefix.map(SmolStr::new),
+                text: text.to_string(),
+            };
+            Some(result)
+        }
+        _ => None,
+    }
+}
+
+fn is_int_number(literal: &ast::Literal) -> bool {
+    match literal.kind() {
+        LiteralKind::IntNumber { .. } => true,
+        _ => false,
+    }
+}
+
+fn remove_separator_from_string(str: &str) -> String {
+    str.replace("_", "")
+}
+
+pub(crate) fn remove_digit_separators(ctx: AssistCtx) -> Option<Assist> {
```

Comment:
Let's put `pub(crate)` functions on top, and add "doc comments" to them (see other assists). These comments then go into auto-generated docs. No need to document *every* assist, but at least a single example would be useful.